perl-lib/OESS/lib/OESS/Cloud/GCP.pm

```diff
@@ -377,15 +377,16 @@ sub insert_interconnect_attachment {
     $req->content( encode_json($payload) );
 
     my $api_response = $http->request($req);
+    my $api_data = decode_json($api_response->content);
```

Comment:
I'm pretty sure that if the response comes back with a 500 code, there will be no json payload. This is the reason for the two separate error checks. I would die in each of the error check if blocks and remove the error key lookup in Cloud.pm.